### ISSUE TITLE
SQLite 3 escape fail

### ISSUE THREAD

**Issue by stoke** [rsdoiel/sqlish#46]
I've noticed that the string escape in SQLite 3 dialect is wrong (it escapes ' as \').

I've also noticed that the right code for escaping is there, but it's commented out. Any plan to fix this?

**Comment by rsdoiel** [comment]
I think my idea was to come up with a unified way to define things escape but I ran out of steam before I implemented what I was thinking about at the time. Let's un-comment it so it work again.

**Comment by rsdoiel** [comment]
Oops, Travis didn't like that. Taking a closer look.

**Comment by rsdoiel** [comment]
Hey Stoke, can you see if this solves the problem.  When you define a dialect you can now define the the escape
sequence to prefix on quotes (third parameter in Sqlish() constructor. The safely() method now checks if this.escape_sequence exists and uses it instead of '\' + c;  Will this get your the quoting you need? I like this approach because as other dialects are supported then they can override the default escape characters as needed.

**Comment by stoke** [comment]
Nope, it doesn't work because `safely`'s context is not set (i.e. this.escape_sequence doesn't exist).

I've came up with a solution based on dialect name, let me know if you want a pr :D

**Comment by rsdoiel** [comment]
Slowly this is coming back to me.  When I create the Sqlish object safely has already been defined and attached
to the global this. That is why I uncommented it and dropped support for SQLite2/3's quoting. Let me sort out the
object constructor. Once I do that I should have something working. I'll add tests this time too :-) so I know if it is solved.

**Comment by rsdoiel** [comment]
Ok, a false tag, a false npm publish and test case later I think I have SQLite3 supported again. Not completely happy with the solution but it'll do until I have the time/energy to rewrite this thing sanely.  Let me know if this works for you. Thanks, Robert
